Library/Homebrew/cmd/audit.rb

```diff
@@ -363,10 +363,16 @@ def audit_deps
           problem "Don't use git as a dependency"
         when "mercurial"
           problem "Use `depends_on :hg` instead of `depends_on 'mercurial'`"
-        when "ruby"
-          problem "Don't use ruby as a dependency. We allow non-Homebrew ruby installations."
         when "gfortran"
           problem "Use `depends_on :fortran` instead of `depends_on 'gfortran'`"
+        when "ruby"
+          problem <<-EOS.undent
+            Don't use "ruby" as a dependency. If this formula requires a
+            minimum Ruby version not provided by the system you should
+            use the RubyRequirement:
```

Comment:
I can't wrap my head around an easy way to necessarily explain `:blah` consistently relates to special Requirement magic sadly. I'm not sure it's an easy concept to grasp if you're unfamiliar with Homebrew's codebase. We've seen enough examples of people doing things like:

```ruby
depends_on "ruby" => "1.8"
```
Before being corrected.